components/size_aware_image.jsx

```diff
@@ -110,18 +115,14 @@ export default class SizeAwareImage extends React.PureComponent {
         return width < MIN_IMAGE_SIZE || height < MIN_IMAGE_SIZE;
     }
 
-    handleLoad = (event) => {
+    handleLoad = () => {
         if (this.mounted) {
-            const image = event.target;
-            const isSmallImage = this.isSmallImage(image.naturalWidth, image.naturalHeight);
             this.setState({
                 loaded: true,
                 error: false,
-                isSmallImage,
-                imageWidth: image.naturalWidth,
```

Comment:
I think we're still going to need to update `this.state.imageWidth` and `this.state.isSmallImage` if `naturalWidth > 0`. Otherwise, we're still relying 100% on the server to provide the dimensions, and if it can't get them for whatever reason, then we're not going to be able to size the image properly after its been downloaded